api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -162,38 +164,56 @@ trait InstitutionsHttpApi extends InstitutionProtocol with ApiErrorProtocol with
   val uploadPath =
     path("institutions" / Segment / "filings" / Segment / "submissions" / Segment) { (institutionId, period, submissionId) =>
       val path = s"institutions/$institutionId/filings/$period/submissions/$submissionId"
-      val uploadTimestamp = Instant.now.toEpochMilli
-      val processingActor = createHmdaRawFile(system, submissionId)
-      processingActor ! StartUpload
-      fileUpload("file") {
-        case (metadata, byteSource) if (metadata.fileName.endsWith(".txt")) =>
-          time {
-            val uploadedF = byteSource
-              .via(splitLines)
-              .map(_.utf8String)
-              .runForeach(line => processingActor ! AddLine(uploadTimestamp, line))
-
-            onComplete(uploadedF) {
-              case Success(response) =>
-                processingActor ! CompleteUpload
-                processingActor ! Shutdown
-                complete {
-                  "uploaded"
-                }
-              case Failure(error) =>
-                processingActor ! Shutdown
-                log.error(error.getLocalizedMessage)
-                val errorResponse = ErrorResponse(400, "Invalid File Format", path)
-                complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
-            }
-          }
+      extractExecutionContext { executor =>
+        val uploadTimestamp = Instant.now.toEpochMilli
+        val processingActor = createHmdaRawFile(system, submissionId)
+        val submissionsActor = system.actorOf(SubmissionPersistence.props(institutionId, period))
+        implicit val ec: ExecutionContext = executor
+        val isNotOverwrite = preventSubmissionOverwrite(submissionsActor, submissionId.toInt)
```

Comment:
I'd like to see a variable name here that indicates that it's a `future`. It seems like lower-case `f` at the start of a variable indicates `future` in other places. So maybe `fCheckForOverwrite` or `fCheckSubmissionStatus`.

OR, you could just circumvent the naming problem with: `onComplete(preventSubmissionOverwrite(submissionsActor, submissionId.toInt))  {`